include/lsst/afw/geom/SphPoint.h

```diff
@@ -0,0 +1,319 @@
+// -*- LSST-C++ -*-
+
+/*
+ * LSST Data Management System
+ * See COPYRIGHT file at the top of the source tree.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program. If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_AFW_GEOM_SPHPOINT_H_
+#define LSST_AFW_GEOM_SPHPOINT_H_
+
+#include <ostream>
+#include <utility>
+
+#include "lsst/afw/geom/Angle.h"
+#include "lsst/afw/geom/Point.h"
+
+namespace lsst {
+namespace afw {
+namespace geom {
+
+    /**
+     * @brief Point in an unspecified spherical coordinate system.
+     *
+     * This class represents a point on a sphere in the mathematical rather
+     * than the astronomical sense; it does not refer to any astronomical
+     * reference frame. Except where otherwise stated, all operations referring
+     * to two or more SphPoint objects assume all points are expressed in the
+     * same coordinate system.
+     *
+     * This class was not designed for inheritance, and must not be subclassed.
+     *
+     * @see @ref coord::Coord
+     */
+    // TODO: add final keyword after migrating to Pybind11
+    class SphPoint
+    {
+    public:
+        /**
+         * @brief Represent a specific pair of spherical coordinates.
```

Comment:
Ok, but I'd leave out the "whose length is ignored", since that's confusing for anybody who isn't familiar with the historical context (when I hear "length" I think "number of elements").